src/ml/neural_net/image_augmentation.hpp

```diff
@@ -269,7 +269,57 @@ class resize_only_image_augmenter final: public image_augmenter {
   options opts_;
 };
 
+/**
+ * An abstract class that inherits from image_augmenter used to convert
+ * input images, annotations and predictions to shared_float_arrays for
+ * tf_image_augmenter.
+ * Subclass must be written for it if needed. The subclass must implement the
+ * pure virtaul method prepare_augmented_images.
+ */
+class float_array_image_augmenter : public image_augmenter {
+ public:
+  float_array_image_augmenter(const options& opts) : opts_(opts) {}
+
+  const options& get_options() const override { return opts_; }
+
+  result prepare_images(std::vector<labeled_image> source_batch) override;
+
+ protected:
+  /** The output sent from TensorFlow after augmenting the images. */
+  struct intermediate_result {
+    /** The images after augmenting sent from Tensorflow */
+    shared_float_array images;
+
+    /** The annotations associated with augmented images sent from Tensorflow */
+    std::vector<shared_float_array> annotations;
+  };
+
+  /** The output sent to TensorFlow to augment the images. */
+  struct intermediate_labeled_image {
```

Comment:
I just double-checked, and I think I gave you bad advice before. Our existing (MPS) implementation ignores the `predictions` entirely. So for now I think it's okay to ignore them here, too. Let's define just one struct, which we can use for both the input and output from the subclass / TensorFlow. Maybe something like `annotated_float_array_image` or `labeled_float_images` or some such?

Reply:
I cannot use the same struct. It is because images member of the struct would be different in both. When we get images as input since their sizes can be different. It is being passed to tensorflow as `std::vector<shared_float_array>`. But when the images are augmented they are also resized in the shape`(416, 416)` and it is being passed back as one `shared_float_array`.

Reply:
Ah, makes sense. Thanks for the explanation!